### PR TITLE
cpu/efm32: model kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description

This adds the Kconfig modules for efm32 and related boards. I added to the CI script one board with EFM32, one with EFR32 and one with EZR32.

### Testing procedure
- Green CI
- You should be able to use the applications that already have a Kconfig configuration with `TEST_KCONFIG=1`.

### Issues/PRs references
#16875

### CODE REVIEW

File: cpu/efm32/Makefile.features [hunk 1]
```diff
@@ -33,3 +33,10 @@ ifeq (1,$(EFM32_LEUART_ENABLED))
 endif
 
 include $(RIOTCPU)/cortexm_common/Makefile.features
+
+# Add efm32 configurations after including cortexm_common so efm32 takes precendence
+# This configuration enables modules that are only available when using Kconfig
+# module modelling
+ifeq (1, $(TEST_KCONFIG))
```
Comment (fjmolinas) [inline cpu/efm32/Makefile.features:40]:
```suggestion
ifeq (1,$(TEST_KCONFIG))
```

File: pkg/cmsis-dsp/Makefile [hunk 2]
```diff
@@ -5,6 +5,9 @@ PKG_LICENSE=Apache-2.0
 
 include $(RIOTBASE)/pkg/pkg.mk
 
+# include makefiles utils tools
+include $(RIOTMAKE)/utils/strings.mk
```
Comment (fjmolinas) [inline pkg/cmsis-dsp/Makefile:9]: Isn't this available early on?

**Review by fjmolinas — COMMENTED** [review]
Thanks for this! Looks good, haven't looked at menuconfig yet..

File: pkg/cmsis-dsp/Makefile (hunk 2 above)
Comment (leandrolanzieri) [reply pkg/cmsis-dsp/Makefile:9]: It was returning empty string, so I'm guessing it's not.

File: cpu/efm32/Makefile.features (hunk 1 above)
Comment (leandrolanzieri) [reply cpu/efm32/Makefile.features]: Fixed

**Comment by fjmolinas** [comment]
please rebase and squash @leandrolanzieri

**Comment by fjmolinas** [comment]
This one needs a rebase @leandrolanzieri !

**Comment by leandrolanzieri** [comment]
Done!

**Review by fjmolinas — COMMENTED** [review]
ACK, will probably need another rebase for the static tests though.

**Review by fjmolinas — APPROVED** [review]
ACK

File: cpu/efm32/periph/Kconfig [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        RTC driver implementation for EFM32 series 0.
+
+config MODULE_PERIPH_RTC_SERIES1
+    bool
+    depends on CPU_EFM32_SERIES1
+    default MODULE_PERIPH_RTC
+        help
+        RTC driver implementation for EFM32 series 1.
+
+config MODULE_PERIPH_RTT_SERIES0
```
Comment (fjmolinas) [inline cpu/efm32/periph/Kconfig:42]: There is a dependency missing here, I think `MODULE_PERIPH_RTT` should select the matching implementation no?

Comment (leandrolanzieri) [reply cpu/efm32/periph/Kconfig:42]: No need to select, these are non-visible symbols with a default. E.g. `MODULE_PERIPH_RTT_SERIES0` depends on being series0, when that is true, then its value will follow `MODULE_PERIPH_RTT`.

File: cpu/efm32/Kconfig [hunk 4]
```diff
@@ -36,15 +39,17 @@ config HAS_CPU_EFM32
...
     default "efm32" if CPU_COMMON_EFM32
 
+config ERROR_MODULES_CONFLICT
+    default "On the EFM32, the RTC and RTT map to the same hardware peripheral." if MODULE_PERIPH_RTC && MODULE_PERIPH_RTT
+    depends on CPU_COMMON_EFM32
+
+menu "EFM32 drivers"
+rsource "drivers/coretemp/Kconfig"
+endmenu
+
```
Comment (leandrolanzieri) [inline cpu/efm32/Kconfig:53]: I think the problem was that I forgot to `rsource "periph/Kconfig"`.

**Comment by leandrolanzieri** [comment]
Ok this is squashed now

**Comment by fjmolinas** [comment]
GO!

**Comment by leandrolanzieri** [comment]
Thanks for the review!

**Pull request merged**